### DISCUSSION TITLE
How can I learn more about this?

### DISCUSSION THREAD

**Discussion by emmyoh** [n0-computer/iroh#144] [Q&A] [answered]
I saw this repository pop up in my feed; there doesn't appear to be anything in the README or anywhere else in this repository describing what this is? Maybe I just haven't looked long/hard enough. I'm curious to know what `iroh` is, exactly. Seeing mentions of P2P, and this is clearly written in Rust, so my curiosity is naturally piqued :)

**Comment by ramfox** [comment]
Hi hi! `iroh` is an opinionated implementation of IPFS, written in rust! We will have a website/docs site up soon. We are in the prototype/research phase, and plan to have a first public release by the end of the year!

Glad to hear you are interested.

**Reply by emmyoh** [reply]
Hope you guys succeed in your efforts! A Rust IPFS implementation would be fantastic and is something I've been hoping for

**Comment by nihalpasham** [comment]
piggy-backing on this question - is there a matrix or discord group for this project? We're exploring `ipfs` for an [`IoT`](https://github.com/nihalpasham/rustBoot) project (involves firmware distribution) and are evaluating (all open-source) rust(y) implementations.

**Reply by b5** [reply]
Apologies for the long gap in response. We're here now 😄 

> is there a matrix or discord group for this project?

At the moment we don't have a discord specifically for iroh, and would like to experiment with using this discussion feature on github for the time being.


> We're exploring ipfs for an [IoT](https://github.com/nihalpasham/rustBoot) project (involves firmware distribution) and are evaluating (all open-source) rust(y) implementations.

I'm hoping you've come across https://github.com/ipfs-rust/ipfs-embed. While the project isn't entirely active it is easily the most robust implementation of IPFS that targets embedded systems at the moment. @rklaehn spearheaded a great deal of the project, and now works with us on iroh. In the long term we expect iroh will also serve this embedded use case, but it will take us at least 10 months to get there with our current plans & engineering bandwidth

**Comment by b5** [comment, marked as answer]
@emmyoh to properly answer your question:

1. The best place to start is https://iroh.computer
2. Discussion happens here, on the github discussion forum!
3. We have a readme now 😄 
4. our [docs home page](https://iroh.computer/docs) describes iroh this way:

> Iroh is a next-generation implementation of the Interplanetary File System ([IPFS](https://ipfs.io/)) for Cloud, Mobile & Desktop platforms. IPFS is a networking protocol for exchanging content-addressed blocks of immutable data. content-addressed means referring to data by the [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of it’s content, which makes the reference both unique and verifiable. These two properties make it possible to get data from any node in the network that speaks the IPFS protocol, including IPFS content being served by other implementations of the protocol.

Hope that helps!
